metadata-validator.py

```diff
@@ -0,0 +1,189 @@
+import json
+import shutil
+import os
+import glob
+import sys
+import subprocess
+
+BUILD_PATH = os.path.abspath(".") + "/.dependency_check/"
+ASN1SOURCES_PATH = os.path.abspath(".") + "/.dependency_check/asn1sources/"
+ERRORLOG_FILE = "errorlog.txt"
+
+METADATA_FILENAME = "meta.json"
+ASN_EXTENSION = ".asn"
+ASN1_EXTENSION = ".asn1"
+ACN_EXTENSION = ".acn"
+
+SUBMODULES_KEY = "submodules"
+ELEMENTS_KEY = "elements"
+REQUIRES_KEY = "requires"
+CONFLICTS_KEY = "conflicts"
+NAME_KEY = "name"
+MODULE_NAME_KEY = "module_name"
+ASN1FILES_KEY = "asn1Files"
+IMPORTS_KEY = "imports"
+FROM_KEY = "from"
+
+MODULE_NAME_KEY = "module_name"
+METADATA_KEY = "meta_json"
+METADATA_PATH_KEY = "meta_path"
+
+ASN1SCC = "asn1.exe"
+ASN1SCC_ARGS = "-ACN -c"
+ASN1SCC_OUT_PATH = BUILD_PATH
+CC = "gcc"
+CC_ARGS = "-c -pipe -O2 -fPIC -Wall -W"
+CC_OUT_PATH = BUILD_PATH
+CC_SRC_PATH = BUILD_PATH
+AR = "ar"
+AR_ARGS = "cqs"
+AR_OUT_PATH = BUILD_PATH
+
+
+def getRecursiveDependency(element_lib, element):
+    dependencies = []
+    if ASN1FILES_KEY in element:
+        for element_filename in element[ASN1FILES_KEY]:
+            dependencies.append(element[METADATA_PATH_KEY] + element_filename)
+
+            acn_filename = element[METADATA_PATH_KEY] + os.path.splitext(element_filename)[0] + ACN_EXTENSION
+            if os.path.exists(acn_filename):
+                dependencies.append(acn_filename)
+    
+    if IMPORTS_KEY in element:
+        for imported in element[IMPORTS_KEY]:
+            for item in element_lib:
+                if ASN1FILES_KEY in item:
+                    if imported[FROM_KEY] in [os.path.splitext(x)[0] for x in item[ASN1FILES_KEY]]:
```

Comment:
for optional keys it might be simpler to use `item.get(ASN1FILES_KEY, [])` where `[]` is the default value returned when no key is found